### PR TITLE
Clean up ROCm4.1 Dockerfile build directory

### PR DESCRIPTION
**Pull request by sunway513**
**Description**: This PR cleans up the build files for ucx, ompi and ort 

**Motivation and Context**
- Why is this change required? What problem does it solve?
The docker images generated with the current Dockerfile is quite large (39 GB). After cleaning those build temps we should be able to reduce the size down to 24 GB. 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/tools/amdgpu/Dockerfile.rocm4.1.pytorch [hunk 1]
```diff
@@ -73,7 +73,8 @@ RUN git clone https://github.com/openucx/ucx.git \
   && cd build \
   && ../contrib/configure-opt --prefix=$UCX_DIR --without-rocm --without-knem --without-cuda \
   && make -j"$(nproc)" \
-  && make install
+  && make install \
+  && make clean
```
Comment (snnn) [inline orttraining/tools/amdgpu/Dockerfile.rocm4.1.pytorch:78]: Why don't delete the whole directory instead?

Comment (sunway513) [reply orttraining/tools/amdgpu/Dockerfile.rocm4.1.pytorch:79]: Done, updated to remove the whole build directories instead.

**Review by snnn — APPROVED** [review]

**Pull request merged**